mypy/plugin.py

```diff
@@ -371,6 +371,23 @@ def lookup_fully_qualified(self, fullname: str) -> Optional[SymbolTableNode]:
         assert self._modules is not None
         return lookup_fully_qualified(fullname, self._modules)
 
+    def get_additional_deps(self, file: MypyFile) -> List[Tuple[int, str, int]]:
+        """Customize dependencies for a module.
+
+        This hook allows adding in new dependencies for a module. It
+        is called after parsing a file but before analysis. This can
+        be useful if a library has dependencies that are dynamic based
+        on configuration information, for example.
+
+        Returns a list of (priority, module name, line number) tuples.
+
+        The line number can be -1 when there is not a known real line number.
+
+        Priorities are defined in mypy.build (but maybe shouldn't be).
+        10 is a good choice for priority.
+        """
```

Comment:
This (+ an example) should be also added to `mypy/docs/source/extending_mypy.rst`.

@msullivan Will you have time to make a PR? I suppose this will be announced in the release blog post, it would be great to have some docs to link to.